src/utils/utils.ts

```diff
@@ -90,13 +123,22 @@ const outputTableFormatter = async (
     ? arrayToAsyncIterable(rowStream)
     : rowStream;
 
+  const encodeRowValues = (row: TableRow) => {
+    for (const key in row) {
+      if (row[key] != null) {
+        // Convert the value to a string and encode non-printable characters
+        row[key] = encodeNonPrintable(row[key].toString());
+      }
+    }
+  };
+
```

Comment:
Why is this an arrow function? It doesn't need to be. Unless you have to use it as a closure then do define it like this. Make it a separate function.

It's a small thing, but each new closure allocates new memory and holds reference to the scope it was created in. It's best to avoid doing it.

Reply:
why are there two loops as well?

Reply:
It's been made into just one loop, and now the `cellLength` rather than the `cellValue` is determined so that extra strings don't need to be instantiated:

```ts
    for (const key in row) {
      if (row[key] != null) {
        row[key] = encodeNonPrintable(row[key].toString());
      }
      // Row[key] is definitely a string or null after this point due to encodeNonPrintable
      const cellValue: string | null = row[key];
      // Null or '' will both cause cellLength to be 3
      const cellLength =
        cellValue == null || cellValue.length === 0 ? 3 : cellValue.length; // 3 is length of 'N/A'
      maxColumnLengths[key] = Math.max(
        maxColumnLengths[key] || 0,
        cellLength, // Use the length of the encoded value
      );
    }
```